packages/runtime/datastore-definitions/src/channel.ts

```diff
@@ -21,13 +21,19 @@ export interface IChannel extends IFluidLoadable {
 
     /**
      * Generates summary of the channel synchronously.
+     * @param fullTree - flag indicating whether the attempt should generate a full
+     * summary tree without any handles for unchanged subtrees.
+     * @param trackState - This tells whether we should track state from this summary.
```

Comment:
If I'm implementing this interface, how do I "trackState"? What does that mean? How can I tell if code does it correctly or not? Thats what this comment needs to answer.

Also from a verbosity standpoint, "This tells whether we should" does not really provide any information, and "we" isn't clear (this is a contract between caller and calee: which is "we"?). I'd prefer something like:

"If true, the implementation must ?????"

Reply:
I believe that for Summaries and also for GC, which leverages the Summary infra,  the default behavior is to always track the state of every object.

@agarwal-navin to make sure it is correct.

Reply:
State tracking is basically an optimization that we added to track state of objects across summaries. Basically, if the state of an object did not change since last successful summary, we can send an `ISummaryHandle` for this object instead of re-summarizing it.
If this is false, the expectation is that you should never send an `ISummaryHandle` since you are not expected to track state.

This was mostly internal and the state tracking happens in summarizer node for data store context and channel context. If a DDS wants to track its internal summary state, it is free to do so (but no DDS does this).

Reply:
There are plans to do this at DDS level - https://dev.azure.com/fluidframework/internal/_workitems/edit/423. When this happens, this option should either be removed or it will make more sense because we would likely expose the previous successful summary. And then we say something like, if `trackState` is true, feel free to use this previous summary to send summary handles.

Reply:
So the requirement here is (and thus a good doc comment would be) - `if false, the implementation must not return a tree containing any ISummaryHandles`?.

Might be better to rename this "allowHandles"?

It would be nice to deduplicate this documentation (as well as `fullTree`) across the various places they occur. This could be done as a non-breaking change by adding a type alias (ex: `export type TrackState = boolean;`) and documenting that, or as a breaking change by switching to an enum, or a summary options object with trackState and fullTree fields document on it.

Reply:
I will leave this change to a different PR.